circuit/types/fullexit_nft.go

```diff
@@ -69,15 +69,21 @@ func SetFullExitNftTxWitness(tx *FullExitNftTx) (witness FullExitNftTxConstraint
 func VerifyFullExitNftTx(
 	api API, flag Variable,
 	tx FullExitNftTxConstraints,
-	accountsBefore [NbAccountsPerTx]AccountConstraints, nftBefore NftConstraints,
+	accountsBefore [NbAccountsPerTx]AccountConstraints,
+	nftBefore NftConstraints,
 ) (pubData [PubDataBitsSizePerTx]Variable) {
+	fromAccount := 0
+	creatorAccount := 1
+
 	pubData = CollectPubDataFromFullExitNft(api, tx)
 	// verify params
-	IsVariableEqual(api, flag, tx.AccountNameHash, accountsBefore[0].AccountNameHash)
-	IsVariableEqual(api, flag, tx.AccountIndex, accountsBefore[0].AccountIndex)
+	IsVariableEqual(api, flag, tx.AccountNameHash, accountsBefore[fromAccount].AccountNameHash)
+	IsVariableEqual(api, flag, tx.AccountIndex, accountsBefore[fromAccount].AccountIndex)
 	IsVariableEqual(api, flag, tx.NftIndex, nftBefore.NftIndex)
 	isCheck := api.IsZero(api.IsZero(tx.CreatorAccountNameHash))
 	isCheck = api.And(flag, isCheck)
+	IsVariableEqual(api, isCheck, tx.CreatorAccountIndex, accountsBefore[creatorAccount].AccountIndex)
```

Comment:
We can use `flag` directly, because now we only allow to deposit L1 nft which is minted by L2 `WithdrawNft`, and this means `tx.CreatorAccountNameHash` can't be empty

Reply:
`isCheck` can be deleted

Reply:
got it, pls check new version